adcm_aio_client/core/objects/cm.py

```diff
@@ -244,10 +253,45 @@ async def hostprovider(self: Self) -> HostProvider:
     def get_own_path(self: Self) -> Endpoint:
         return self.PATH_PREFIX, self.id
 
+    def __str__(self: Self) -> str:
+        return f"<{self.__class__.__name__} #{self.id} {self.name}>"
+
 
-class HostsNode(PaginatedAccessor[Host, None]):
+class HostsAccessor(PaginatedAccessor[Host, dict | None]):
     class_type = Host
 
 
-class HostsInClusterNode(PaginatedAccessor[Host, None]):
-    class_type = Host
+class HostsInClusterNode(HostsAccessor):
+    async def add(self: Self, host: Host | Iterable[Host] | None = None, filters: Filter | None = None) -> None:
+        hosts = await self._get_hosts_from_arg_or_filter(host=host, filters=filters)
+
+        await self._requester.post(*self._path, data=[{"hostId": host.id} for host in hosts])
+
+    async def remove(self: Self, host: Host | Iterable[Host] | None = None, filters: Filter | None = None) -> None:
+        hosts = await self._get_hosts_from_arg_or_filter(host=host, filters=filters)
+
+        results = await asyncio.gather(
+            *(self._requester.delete(*self._path, host_.id) for host_ in hosts), return_exceptions=True
```

Comment:
generators can't be unpacked like that

gather doesn't take iterable?

Reply:
why? it works. gather takes sequence of awaitables